### PR TITLE
Add a page for checking payments via the tx key

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
Needshttps://github.com/monero-project/monero/pull/1297

File: src/libwalletqt/WalletManager.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "WalletManager.h"
 #include "Wallet.h"
 #include "wallet/wallet2_api.h"
+#include <boost/lexical_cast.hpp>
```
Comment (Jaqueeee) [inline src/libwalletqt/WalletManager.cpp]: Gives build error (os x). Had to add path to boost to monero-core.pro
Not sure if that is needed on other platforms.

File: src/libwalletqt/WalletManager.cpp [hunk 2]
```diff
@@ -164,6 +165,16 @@ QString WalletManager::paymentIdFromAddress(const QString &address, bool testnet
     return QString::fromStdString(Bitmonero::Wallet::paymentIdFromAddress(address.toStdString(), testnet));
 }
 
+QString WalletManager::checkPayment(const QString &address, const QString &txid, const QString &txkey, const QString &daemon_address) const
+{
+    uint64_t received = 0, height = 0;
+    std::string error = "";
+    bool ret = m_pimpl->checkPayment(address.toStdString(), txid.toStdString(), txkey.toStdString(), daemon_address.toStdString(), received, height, error);
+    // bypass qml being unable to pass structures without preposterous complexity
+    std::string result = std::string(ret ? "true" : "false") + "|" + boost::lexical_cast<std::string>(received) + "|" + boost::lexical_cast<std::string>(height) + "|" + error;
```
Comment (Jaqueeee) [inline src/libwalletqt/WalletManager.cpp]: nice!

**Comment by moneromooo-monero** [comment]
fixed

File: LeftPanel.qml [hunk 3]
```diff
@@ -256,6 +258,21 @@ Rectangle {
                 }
             }
 
+            // ------------- TxKey tab ---------------
+            MenuButton {
+                id: txkeyButton
+                anchors.left: parent.left
+                anchors.right: parent.right
+                text: qsTr("Tx Key") + translationManager.emptyString
```
Comment (Jaqueeee) [inline LeftPanel.qml]: Maybe change this to "Check payment"?

**Comment by medusadigital** [comment]
tested negative and positive scenarios, worked always. 
also tested with remotenodes.

did not test testnet.

**Pull request merged**